### DISCUSSION TITLE
Ive implemented XCTest.framework but getting 'Not authorized for performing UI testing actions.' on synthesizeWithError

### DISCUSSION THREAD

**Discussion by JohnHDev** [appium/appium#19801] [Q&A]
Hi, this isn't an issue with Appium specifically, but I was following the instructions here:
https://appium.github.io/appium-xcuitest-driver/5.11/actions/

Ive followed the instructions to simulate a tap but I get an error back when I try to synthesize. Calling synthesizeWithError results in an NSError containing 'Not authorized for performing UI testing actions.'

Any ideas what I might be doing wrong? What do I need to do to make the iOS app authorised?
My google-fu has failed me so far, Im running on an iOS simulator with Xcode 15.1.

Thanks
John

**Comment by portyu9** [comment]
Hi @JohnHDev 

There is no entitlement/capability you can add to a normal iOS application to authorize `synthesizeWithError:`.

`XCPointerEventPath` and `XCSynthesizedEventRecord` are private XCTest APIs and must run in an authorized XCTest UI-test runner context. Simply linking `XCTest.framework` into the app is not sufficient.

If your goal is automation, let Appium/WebDriverAgent perform the gesture instead:

```java
driver.executeScript("mobile: tap", Map.of(
    "x", 100,
    "y", 200
));
```

For more complex gestures, use W3C Actions.

If you are implementing this yourself instead of using Appium, it must execute from an XCTest **UI Testing target/runner**, not from the application under test.

Please mark this answer as accepted if it helped, thank you.
